Sources/XcbeautifyLib/Renderers/GitHubActionsRenderer.swift

```diff
@@ -0,0 +1,279 @@
+        return outputGitHubActionsLog(
+            annotationType: .error,
+            message: line
+        )
+    }
+
+    func formatRestartingTest(line: String, group: RestartingTestCaptureGroup) -> String {
+        let message = Format.indent + line
+        return outputGitHubActionsLog(
+            annotationType: .error,
+            message: message
+        )
+    }
+
+    func formatTestCaseMeasured(group: TestCaseMeasuredCaptureGroup) -> String {
+        let testCase = group.testCase
+        let name = group.name
+        let unitName = group.unitName
+        let deviation = group.deviation
+        let value = group.value
+
+        return Format.indent + testCase + " measured (\(value) \(unitName) ±\(deviation)% -- \(name))"
+    }
+
+    func formatTestCasePassed(group: TestCasePassedCaptureGroup) -> String {
+        let testCase = group.testCase
+        let time = group.time
+        return Format.indent + testCase + " (\(time) seconds)"
+    }
+
+    func formatUIFailingTest(group: UIFailingTestCaptureGroup) -> String {
+        let file = group.file
+        let fileComponents = file.asFileComponents()
+        let failingReason = group.reason
+        let message = Format.indent + failingReason
+        return outputGitHubActionsLog(
+            annotationType: .error,
+            fileComponents: fileComponents,
+            message: message
+        )
+    }
+
+    func formatWarning(group: GenericWarningCaptureGroup) -> String {
+        let warningMessage = group.wholeWarning
+        return outputGitHubActionsLog(
+            annotationType: .warning,
+            message: warningMessage
+        )
+    }
+
+    func formatWillNotBeCodesignWarning(group: WillNotBeCodeSignedCaptureGroup) -> String {
+        let warningMessage = group.wholeWarning
+        return outputGitHubActionsLog(
+            annotationType: .warning,
+            message: warningMessage
+        )
+    }
+}
+
+extension GitHubActionsRenderer {
+    func formatTestSummary(isSuccess: Bool, description: String) -> String {
+        if isSuccess {
+            return outputGitHubActionsLog(
+                annotationType: .notice,
+                message: "Tests Passed: \(description)"
+            )
+        } else {
+            return outputGitHubActionsLog(
+                annotationType: .error,
+                message: "Tests Failed: \(description)"
+            )
+        }
+    }
+}
+
+
+private struct FileComponents {
+    private let path: String
+    private let line: Int?
+    private let column: Int?
+
+    init(path: String, line: Int?, column: Int?) {
+        self.path = path
+        self.line = line
+        self.column = column
+    }
+
+    var formatted: String {
+        guard let line = line else {
+            return "file=\(path)"
+        }
+
+        guard let column = column else {
+            return "file=\(path),line=\(line)"
+        }
+
+        return "file=\(path),line=\(line),col=\(column)"
+    }
+}
```

Comment:
What do you think about pulling this type out into a separate file, even though it's only used here in this class? That might make it easier to find in the future.

Reply:
I'd prefer to keep it in this file, since this is specifically designed to only be used with the `GitHubActionsRenderer`.